Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /home/vadim/Projects/WORK_PRO/TASK/ADD_NEW_DATA/t2/shm_class_in_class_in_map/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_528e5/fast && /usr/bin/make  -f CMakeFiles/cmTC_528e5.dir/build.make CMakeFiles/cmTC_528e5.dir/build
make[1]: Entering directory '/home/vadim/Projects/WORK_PRO/TASK/ADD_NEW_DATA/t2/shm_class_in_class_in_map/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_528e5.dir/src.c.o
/usr/bin/cc   -DCMAKE_HAVE_LIBC_PTHREAD -o CMakeFiles/cmTC_528e5.dir/src.c.o -c /home/vadim/Projects/WORK_PRO/TASK/ADD_NEW_DATA/t2/shm_class_in_class_in_map/build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_528e5
/home/vadim/.local/lib/python2.7/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_528e5.dir/link.txt --verbose=1
/usr/bin/cc  -DCMAKE_HAVE_LIBC_PTHREAD CMakeFiles/cmTC_528e5.dir/src.c.o -o cmTC_528e5 
/usr/bin/ld: CMakeFiles/cmTC_528e5.dir/src.c.o: in function `main':
src.c:(.text+0x46): undefined reference to `pthread_create'
/usr/bin/ld: src.c:(.text+0x52): undefined reference to `pthread_detach'
/usr/bin/ld: src.c:(.text+0x5e): undefined reference to `pthread_cancel'
/usr/bin/ld: src.c:(.text+0x6f): undefined reference to `pthread_join'
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_528e5.dir/build.make:106: cmTC_528e5] Error 1
make[1]: Leaving directory '/home/vadim/Projects/WORK_PRO/TASK/ADD_NEW_DATA/t2/shm_class_in_class_in_map/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:140: cmTC_528e5/fast] Error 2


Source file was:
#include <pthread.h>

void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/vadim/Projects/WORK_PRO/TASK/ADD_NEW_DATA/t2/shm_class_in_class_in_map/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_84e3c/fast && /usr/bin/make  -f CMakeFiles/cmTC_84e3c.dir/build.make CMakeFiles/cmTC_84e3c.dir/build
make[1]: Entering directory '/home/vadim/Projects/WORK_PRO/TASK/ADD_NEW_DATA/t2/shm_class_in_class_in_map/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_84e3c.dir/CheckFunctionExists.c.o
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create -o CMakeFiles/cmTC_84e3c.dir/CheckFunctionExists.c.o -c /home/vadim/.local/lib/python2.7/site-packages/cmake/data/share/cmake-3.18/Modules/CheckFunctionExists.c
Linking C executable cmTC_84e3c
/home/vadim/.local/lib/python2.7/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_84e3c.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create CMakeFiles/cmTC_84e3c.dir/CheckFunctionExists.c.o -o cmTC_84e3c  -lpthreads 
/usr/bin/ld: cannot find -lpthreads
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_84e3c.dir/build.make:106: cmTC_84e3c] Error 1
make[1]: Leaving directory '/home/vadim/Projects/WORK_PRO/TASK/ADD_NEW_DATA/t2/shm_class_in_class_in_map/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:140: cmTC_84e3c/fast] Error 2



